Dart/src/com/jetbrains/lang/dart/ide/completion/DartServerCompletionContributor.java

```diff
@@ -156,9 +156,9 @@ protected void addCompletions(@NotNull final CompletionParameters parameters,
                    for (Map.Entry<String, Map<String, Set<String>>> entry : existingImports.entrySet()) {
                      String importedLibraryUri = entry.getKey();
                      Map<String, Set<String>> importedLibrary = entry.getValue();
-                     // Checks whether any of the libraries exported by this import declares the same name as this suggestion.
-                     if (importedLibrary.entrySet().stream().anyMatch(
-                         importEntry -> doesImportSuggestion(importEntry, suggestionSet.getUri(), suggestion.getLabel()))) {
+                     // TODO: Use individual suggestion uris instead of suggestion set uri.
```

Comment:
DAS does provide `AvailableSuggestion.declaringLibraryUri` with https://dart-review.googlesource.com/c/sdk/+/104808
Is there something that prevents us from using it in this PR?

Reply:
Nope, I just pushed this before you landed that change :). Updating now!